hera_cal/io.py

```diff
@@ -1829,3 +1830,63 @@ def baselines_from_filelist_position(filename, filelist):
     upper_index = np.min([(file_index + 1) * chunk_size, nbls])
     output = bls[lower_index:upper_index]
     return output
+
+
+def throw_away_flagged_ants(infilename, outfilename, yaml_file=None, throw_away_fully_flagged_data_baselines=False, clobber=False):
+    """Throw away completely flagged data.
+
+    Parameters
+    ----------
+        infilename: str
+            path to a UVData file in uvh5 format.
+        outfilename: str
+            path to file to output trimmed data file.
+        yaml_file: str
+            path to a yaml flagging file with a list of antennas to flag.
+            Default is None.
+        throw_away_flagged_ants: bool, optional
+            if True, also throw away baselines where all data is flagged.
+            Warning: Don't use this for files with a small number of time integrations.
+                     since this can easily happen by chance in files like this.
+            Default is False
+        clobber: bool, optional
+            overwrite output file if it already exists.
+            Default is False.
+    Returns
+    -------
+        N/A
+
+    """
+    hd = HERAData(infilename)
+    hd.read()
+
+    # throw away flagged antennas in yaml file.
+    if yaml_file is not None:
+        from hera_qm import utils as qm_utils
+        qm_utils.apply_yaml_flags(uv=hd, a_priori_flag_yaml=yaml_file,
+                                  ant_indices_only=True, flag_ants=True, flag_freqs=False,
+                                  flag_times=False, throw_away_flagged_ants=True)
+
+    # Write data
+    if throw_away_fully_flagged_data_baselines:
+        antpairs_to_keep = []
+        for antpair in hd.get_antpairs():
+            fully_flagged = True
+            for pol in hd.pols:
+                fully_flagged = fully_flagged & np.all(hd.get_flags(antpair + (pol, )))
+            if not fully_flagged:
+                antpairs_to_keep.append(antpair)
+        hd.select(bls=antpairs_to_keep)
+    hd.write_uvh5(outfilename, clobber=clobber)
```

Comment:
Can I suggest adding something to the history explaining which baselines were thrown out?

Reply:
added history string.